File: springboot-version/backend/src/main/java/com/petconnect/controller/AuthController.java
package com.petconnect.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.petconnect.dto.LoginRequest;
import com.petconnect.dto.OtpVerifyRequest;
import com.petconnect.dto.RegisterRequest;
import com.petconnect.service.UserService;

import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;

@Validated
@RestController
@RequestMapping("/api/auth")
@RequiredArgsConstructor
public class AuthController {

    private final UserService userService;

    @PostMapping("/register")
    public ResponseEntity<?> register(@RequestBody @Valid RegisterRequest request) {
        return ResponseEntity.ok(userService.register(request));
    }

    @PostMapping("/login")
    public ResponseEntity<?> login(@RequestBody @Valid LoginRequest request) {
        return ResponseEntity.ok(userService.login(request));
    }

    @PostMapping("/verify-otp")
    public ResponseEntity<?> verifyOtp(@RequestBody @Valid OtpVerifyRequest request) {
        return ResponseEntity.ok(userService.verifyOtp(request));
    }

}

File: aspnet-version/backend/PetConnectAlmost/Controllers/AuthController.cs
using BCrypt.Net;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;
using PetConnectAlmost.Data;
using PetConnectAlmost.Models;
using PetConnectAlmost.Services;

namespace PetConnectAlmost.Controllers

{
    [ApiController]
    [Route("api/auth")]
    public class AuthController:ControllerBase
    {
        private readonly IAuthService authService;

        public AuthController(IAuthService authService)
        {
            this.authService = authService;
        }

        [HttpGet("ping")]
        public IActionResult Ping() => Ok("Auth API is reachable!");
        [HttpPost("register")]
        public async Task<IActionResult> Register(RegisterRequest request)
        {
            var result = await authService.RegisterAsync(request);
            if (result is ErrorResponse error && error.StatusCode >= 400)
            {
                return StatusCode(error.StatusCode, error);
            }

            return Ok(result);

        }
        [HttpPost("login")]
        public async Task<IActionResult> Login(LoginRequest dto)
        {
            var result = await authService.LoginAsync(dto);
            if (result is ErrorResponse error && error.StatusCode >= 400)
            {
                return StatusCode(error.StatusCode, error);
            }

            return Ok(result);
        }

        [HttpPost("verify-otp")]
        public async Task<IActionResult> VerifyOtp(OtpVerifyRequest request)
        {
            var result = await authService.VerifyOtpAsync(request);
            if (result is ErrorResponse error && error.StatusCode >= 400)
            {
                return StatusCode(error.StatusCode, error);
            }

            return Ok(result);
        }

    }
}
